Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -2580,4 +2580,12 @@ - (void)performMultipleWatchStreamsTest:(nullable dispatch_queue_t)delegateQueue
     [self waitForExpectations:@[expectation] timeout:60.0];
 }
 
+- (void)testSyncConfigShouldNotMigrate {
+    RLMUser *user = [self userForTest:_cmd];
+    NSString *realmId = self.appId;
+    RLMRealm *realm = [self openRealmForPartitionValue:realmId user:user];
+
+    XCTAssertThrows([realm.configuration setDeleteRealmIfMigrationNeeded:YES], @"Not allow 'deleteRealmIfMigrationNeeded' when sync is enabled.");
```

Comment:
This test should also test that valid configurations still work, even though it is tested elsewhere. Should also write on for swift (why not?).

Reply:
Ok